pkg/lib/cockpit-components-table.tsx

```diff
@@ -65,11 +69,60 @@ const _ = cockpit.gettext;
  *   event, isSelected, rowIndex, rowData, extraData. rowData contains props with an id property of the clicked row.
  * - onHeaderSelect: event, isSelected.
  */
+
+interface ListingTableRowColumnProps {
+    title: React.ReactNode;
+    sortKey?: string;
+    props?: TdProps | ThProps;
+}
+
+type ListingTableRowColumn = React.ReactNode | ListingTableRowColumnProps;
+
+function cell_is_props(cell: ListingTableRowColumn): cell is ListingTableRowColumnProps {
+    return typeof cell == "object" && (cell as ListingTableRowColumnProps).title !== undefined;
+}
+
+export interface ListingTableRowProps {
+    columns: ListingTableRowColumn[];
+    props?: TrProps;
+    expandedContent?: React.ReactNode;
+    selected?: boolean;
+    initiallyExpanded?: boolean;
+    hasPadding?: boolean;
+}
+
+export interface ListingTableColumnProps {
+    title: string;
+    header?: boolean;
+    sortable?: boolean;
+    props?: ThProps;
+}
+
+export interface ListingTableProps extends Omit<TableProps, 'rows' | 'onSelect'> {
+    actions?: React.ReactNode[],
+    afterToggle?: (expanded: boolean) => void,
+    caption?: string,
+    className?: string,
+    columns: (string | ListingTableColumnProps)[],
```

Comment:
Should we just deprecate `string[]`?

Reply:
I would vote "no". It doesn't cause any problems and is convenient to use, no?